LaithornsGrace/src/main/java/com/dreamless/laithorn/player/PlayerDataHandler.java

```diff
@@ -4,14 +4,19 @@
 
 public class PlayerDataHandler {
 
-	public static int LEVEL_ONE_EXP = 150;
-	public static double GROWTH_RATE = 1.17; // ~7% increase
-	public static int LEVEL_CAP = 99;
-
-	// Note, level 99 needs 1,795,670 exp
+	private static int LEVEL_ONE_EXP = 7500;
+	private static double GROWTH_RATE = 1.75;
+	private static int LEVEL_CAP = 10;
 
 	public static int getNewEXPRequirement(int level) {
-		return (int) Math.round(LEVEL_ONE_EXP * Math.pow(GROWTH_RATE, level));
+		if(level >= LEVEL_CAP)
+		{
+			return 0;
```

Comment:
I think the loop that calls this would go infinite if it hits this case -- iirc it does `while (remaining > 0) remaining -= getNewEXP` so it would never exit the loop if getNewEXP returned zero

Reply:
Thanks for catching that. I'm doing a level cap check at the end and doing a break.